src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/component/SNSPublisher.kt

```diff
@@ -1,25 +1,38 @@
 package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.component
 
 import com.fasterxml.jackson.databind.ObjectMapper
+import com.microsoft.applicationinsights.TelemetryClient
 import mu.KLogging
 import org.springframework.beans.factory.annotation.Value
 import org.springframework.stereotype.Component
 import software.amazon.awssdk.services.sns.SnsClient
 import software.amazon.awssdk.services.sns.model.PublishRequest
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.dto.EventDTO
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.AuthUser
+import java.util.UUID
 
 @Component
 class SNSPublisher(
   private val client: SnsClient,
   private val objectMapper: ObjectMapper,
+  private val telemetryClient: TelemetryClient,
   @Value("\${aws.sns.enabled}") private val enabled: Boolean,
   @Value("\${aws.sns.topic.arn}") private val arn: String,
 ) {
   companion object : KLogging()
 
-  fun publish(event: EventDTO) {
+  fun publish(referralId: UUID, actor: AuthUser?, event: EventDTO) {
```

Comment:
nit: non nullable actor would be nice

Reply:
yeah, we can do that as soon as we add the `appointmentAttendedRecordedBy` and similar "by" fields on appointment